### PR TITLE
PAPP-21340: Add submit_file action

### PR DESCRIPTION
**Pull request by achandya-crest**
Integration pipeline link: https://cd.splunkdev.com/phantom-apps/app-tests/-/pipelines/4354488

Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [x] Testing of all the changes has been performed (for bug fixes / features)
- [x] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [x] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [x] Verify all checks are passing.
- [x] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [x] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

File: zscaler_connector.py [hunk 1]
```diff
@@ -793,39 +874,44 @@ def finalize(self):
...
     args = argparser.parse_args()
+    verify = args.verify
     session_id = None
 
     if (args.username and args.password):
         login_url = BaseConnector._get_phantom_base_url() + "login"
         try:
             print("Accessing the Login page")
-            r = requests.get(login_url, verify=False)
+            r = requests.get(    # nosemgrep: python.requests.best-practice.use-timeout.use-timeout
```
Comment (ricardo-phantom) [inline zscaler_connector.py:897]: remove blank space

File: zscaler_connector.py [hunk 2]
```diff
@@ -793,39 +874,44 @@ def finalize(self):
...
+                login_url, verify=verify)
             csrftoken = r.cookies['csrftoken']
             data = {'username': args.username, 'password': args.password, 'csrfmiddlewaretoken': csrftoken}
             headers = {'Cookie': 'csrftoken={0}'.format(csrftoken), 'Referer': login_url}
 
             print("Logging into Platform to get the session id")
-            r2 = requests.post(login_url, verify=False, data=data, headers=headers)
+            r2 = requests.post(    # nosemgrep: python.requests.best-practice.use-timeout.use-timeout
```
Comment (ricardo-phantom) [inline zscaler_connector.py:904]: remove blank space

Comment (achandya-crest) [reply zscaler_connector.py]: I removed 2 additional spaces before inline comment. Now there are only 2 spaces left that are necessary to be added before an inline comment.

File: zscaler_connector.py (hunk 1 above)
Comment (achandya-crest) [reply zscaler_connector.py]: I removed 2 additional spaces before inline comment. Now there are only 2 spaces left that are necessary to be added before an inline comment.

**Review by ricardo-phantom — APPROVED** [review]

File: zscaler_view.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+def get_ctx_result(result):
+
+    ctx_result = {}
+    param = result.get_param()
+    summary = result.get_summary()
+    data = result.get_data()
+
+    ctx_result['param'] = param
+
+    if (data):
```
Comment (rgil-splunk) [inline zscaler_view.py:24]:
```suggestion
    if data:
```

File: zscaler_view.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+
+def display_view(provides, all_app_runs, context):
+
+    context['results'] = results = []
+    for summary, action_results in all_app_runs:
+        for result in action_results:
+
+            ctx_result = get_ctx_result(result)
+            if (not ctx_result):
```
Comment (rgil-splunk) [inline zscaler_view.py:40]:
```suggestion
            if not ctx_result:
```

File: zscaler_view.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    param = result.get_param()
+    summary = result.get_summary()
+    data = result.get_data()
+
+    ctx_result['param'] = param
+
+    if (data):
+        ctx_result['data'] = data[0]
+
+    if (summary):
```
Comment (rgil-splunk) [inline zscaler_view.py:27]:
```suggestion
    if summary:
```

**Review by rgil-splunk — APPROVED** [review]

File: zscaler_view.py (hunk 5 above)
Comment (achandya-crest) [reply zscaler_view.py]: Done

File: zscaler_view.py (hunk 4 above)
Comment (achandya-crest) [reply zscaler_view.py]: Done

File: zscaler_view.py (hunk 3 above)
Comment (achandya-crest) [reply zscaler_view.py]: Done

**Pull request merged**